### ISSUE TITLE
Thread Pool size

### ISSUE THREAD

**Issue by PedroMPagani** [lucko/helper#136]
Hey, the async option from the Executors doesn't limit the number of threads that can be used in the pool, users have to be very careful when using that to not get an out of memory, a proper usage of it would be to limit the number of threads.

**Comment by Baricaade** [comment]
If I’m not mistaken helper’s asynchronous executor uses a cached thread pool which should remove unused threads after a certain period of time. So unless you’re running an extremely large number of long and intensive tasks, it should be fine.

**Comment by PedroMPagani** [comment]
You're not mistaken, that is correct, but when executing many things at the same time will end up in our of memory, don't you think that there should be a max limit? Specially for developers, people tend to think that just running async will fix any lag issue.

**Comment by Conclure** [comment]
Perhaps just give callers the choice of dropping their own executor instance if wanted to? Similarly to how CompletableFuture allows it with its factory methods CompletableFuture::runAsync and CompletableFuture::supplyAsync.

**Comment by PedroMPagani** [comment]
That sounds more like an open idea, good thinking!
